src/tools.ts

```diff
@@ -150,28 +135,36 @@ export function convertMcpToolToLangchainTool(
  */
 export async function loadMcpTools(
   client: Client,
-  responseFormat: ResponseFormat = 'text'
+  responseFormat: ResponseFormat = 'content',
+  throwOnLoadError: boolean = true
 ): Promise<StructuredToolInterface[]> {
-  try {
-    // Get tools in a single operation
-    const toolsResponse = await client.listTools();
-    logger.info(`Found ${toolsResponse.tools?.length || 0} MCP tools`);
-
-    // Filter out tools without names and convert in a single map operation
-    return (toolsResponse.tools || [])
-      .filter(tool => !!tool.name)
-      .map(tool => {
-        try {
-          logger.debug(`Successfully loaded tool: ${tool.name}`);
-          return convertMcpToolToLangchainTool(client, tool, undefined, responseFormat);
-        } catch (error) {
-          logger.error(`Failed to load tool "${tool.name}":`, error);
-          return null;
+  // Get tools in a single operation
+  const toolsResponse = await client.listTools();
+  logger.info(`Found ${toolsResponse.tools?.length || 0} MCP tools`);
+
+  // Filter out tools without names and convert in a single map operation
+  return (toolsResponse.tools || [])
+    .filter(tool => !!tool.name)
+    .map(tool => {
+      try {
+        const dst = new DynamicStructuredTool({
+          name: tool.name,
+          description: tool.description || '',
+          schema: JSONSchemaToZod.convert(
+            (tool.inputSchema ?? { type: 'object', properties: {} }) as JSONSchema
+          ),
```

Comment:
This was the main thing that was missing previously. I also simplified the way tools are being built.